src/Database/Relations/BelongsToMany.php

```diff
@@ -121,14 +123,11 @@ public function attach($id, array $attributes = [], $touch = true)
          *     $model->bindEvent('model.relation.beforeAttach', function (string $relationName, array $attachedIdList, array $insertData) use (\October\Rain\Database\Model $model) {
          *         if (!$model->isRelationValid($attachedIdList)) {
          *             throw new \Exception("Invalid relation!");
-         *             return false;
          *         }
          *     });
          *
          */
-        if ($this->parent->fireEvent('model.relation.beforeAttach', [$this->relationName, $attachedIdList, $insertData], true) === false) {
-            return;
-        }
+        $this->parent->fireEvent('model.relation.beforeAttach', [$this->relationName, $attachedIdList, $insertData]);
 
         // Here we will insert the attachment records into the pivot table. Once we have
```

Comment:
**Question:** Detach method calls parent::detach(), great. But why attach method doesn't? Some [parent code](https://github.com/laravel/framework/blob/6.x/src/Illuminate/Database/Eloquent/Relations/Concerns/InteractsWithPivotTable.php#L249-L251) is bypassed. I know that it wasn't changed by @mjauvin, so only asking...

Reply:
You're definitely right here, `parent::attach()` has changed in Laravel 6 and these changes are lost because of this. We should definitely call `parent::attach()`